.github/scripts/main/main.sh

```diff
@@ -48,6 +48,49 @@ make install
 export PATH="$PREFIX/bin:$PATH"
 opam --version
 
+if [[ $OPAM_DOC -eq 1 ]]; then
+  make -C doc html man-html pages
+
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
+    . .github/scripts/common/hygiene-preamble.sh
+    diff="git diff $BASE_REF_SHA..$PR_REF_SHA"
+    set +e
+    files="$($diff --name-only --diff-filter=A -- src/**/*.mli)"
+    set -e
+    if [ -n "$files" ]; then
+      echo '::group::new module(s) added - check index updated too'
+      if $diff --name-only --exit-code -- doc/index.html ; then
+        echo '::error new module(s) added but doc/index.html not updated'
+        echo "$files"
+        exit 3
+      fi
+      echo '::endgroup::new module added - checking it'
+    else
+      echo 'No new modules added'
+    fi
+  fi
+
+  htmlfiles=$(ls doc/pages/*.md | sed -e 's/\.md$/.html/')
+  manfiles=$(opam help topics | sed -e 's|.*|doc/man-html/opam-&.html|')
+  manfiles="$manfiles $(opam admin help topics | sed -e 's|.*|doc/man-html/opam-admin-&.html|')"
+
+  echo '::group::checking for generated files'
+  echo "pages: $htmlfiles"
+  echo "topics: $manfiles"
+  files="$htmlfiles $manfiles"
+  missing=""
+  for file in $files; do
```

Comment:
It doesn't matter much here, but in general it is better practice to use arrays instead of strings to encode a list of things. This allows support for files with spaces and other special characters. Luckily we're using bash here and not POSIX sh so we can actually use this feature ^^
```suggestion
  mapfile -t htmlfiles < <(ls doc/pages/*.md | sed -e 's/\.md$/.html/')
  mapfile -t manfiles < <(opam help topics | sed -e 's|.*|doc/man-html/opam-&.html|')
  mapfile -O "${#manfiles[@]}" -t manfiles < <(opam admin help topics | sed -e 's|.*|doc/man-html/opam-admin-&.html|')

  echo '::group::checking for generated files'
  echo "pages: $htmlfiles"
  echo "topics: $manfiles"
  files=("${htmlfiles[@]}" "${manfiles[@]")
  missing=""
  for file in "${files[@]}"; do
```